cleartk-berkeleyparser/src/main/java/org/cleartk/berkeleyparser/ParserAnnotator.java

```diff
@@ -55,14 +59,17 @@
 
 public class ParserAnnotator<TOKEN_TYPE extends Annotation, SENTENCE_TYPE extends Annotation, TOP_NODE_TYPE extends Annotation>
     extends ParserWrapper_ImplBase<TOKEN_TYPE, SENTENCE_TYPE, Tree<String>, TOP_NODE_TYPE> {
-  
-  public static AnalysisEngineDescription getDescription(String modelPath) throws ResourceInitializationException {
+
+  public static AnalysisEngineDescription getDescription(String modelPath)
+      throws ResourceInitializationException {
     return AnalysisEngineFactory.createEngineDescription(
         ParserAnnotator.class,
         ParserAnnotator.PARAM_PARSER_MODEL_PATH,
         modelPath,
         ParserWrapper_ImplBase.PARAM_OUTPUT_TYPES_HELPER_CLASS_NAME,
-        DefaultOutputTypesHelper.class.getName());
+        DefaultOutputTypesHelper.class.getName(),
+        ParserWrapper_ImplBase.PARAM_TOKENIZER_CLASS_NAME,
```

Comment:
@mjlaali - It is more UIMA-like to make the tokenizer a separate annotator which adds tokens to the CAS.  This way the ParserAnnotator can just focus on extracting what it needs from the CAS independently of how tokenization and other preprocessing was performed.

For examples of this decoupling, I would look at our [ClearNLP wrappers](https://github.com/ClearTK/cleartk/tree/master/cleartk-clearnlp/src/main/java/org/cleartk/clearnlp).